packages/video_player/closed_caption_file/lib/src/sub_rip.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../closed_caption_file.dart';
+
+/// This is a method that takes in a string (in the SubRip file format) and
+/// produces a list of [Caption]s.
+///
+/// This method should not be used directly, instead, use
+/// `ClosedCaptionFile.fromSubRipFile(...)`.
+List<Caption> parseCaptionsFromSubRipString(String file) {
+  final captions = <Caption>[];
```

Comment:
Whoops, this and the various `var` definitions throughout here are passing our linter. I thought that was an automated check but it looks like we don't actually have it on right now. I'll put in a PR to turn [always_specify_types](https://dart-lang.github.io/linter/lints/always_specify_types.html) on. In the meantime you may want to manually patch that in your environment and that give this PR another look with it on.

```
  info • Specify type annotations at lib/src/sub_rip.dart:15:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:16:8 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:19:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:20:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:22:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:24:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:47:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:49:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:50:5 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:63:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:64:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:66:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:67:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:68:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:69:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:82:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:84:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:85:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:86:3 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:87:8 • (always_specify_types)
  info • Specify type annotations at lib/src/sub_rip.dart:88:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:13:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:14:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:19:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:25:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:37:5 • (always_specify_types)
  info • Specify type annotations at test/sub_rip_file_test.dart:52:5 • (always_specify_types)
```

Reply:
Ah, didn't realize that lint was enforced in this package. I think I fixed all the violations.